_articles/network.md

```diff
@@ -0,0 +1,313 @@
+-   **Layer 2:** Data Link
+-   **Layer 1:** Physical
+
+We will assume that you are familiar with this model, and will proceed by discussing ways to troubleshoot issues at stack Layers 1 through 4.
+Where to start troubleshooting is situation-dependent. For example, if you can SSH to a server, but the server can't connect to a MySQL database, the problem is unlikely to be the physical or data link layers
+on the local server. In general, it's a good idea to work your way down the stack. Start with the application, and then gradually troubleshoot each lower layer until you've isolated the problem.
+
+With that background out of the way, let's jump to the command line and start troubleshooting.
+
+[]{#anchor-1}Layer 1: The physical layer
+----------------------------------------
+
+We often take the physical layer for granted (\"did you make sure the cable is plugged in?\"), but we can easily troubleshoot physical layer problems from the Linux command line. That is if you have console connectivity to the host, which might not be the case for some remote systems.
+
+Let's start with the most basic question: Is our physical interface up?
+The **ip link show** command tells us:
+
+****\#*** *ip link show**
+
+**1: lo: \<LOOPBACK,UP,LOWER\_UP\> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000**
+**link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00**
+
+**2:* *eth0: \<BROADCAST,MULTICAST,DOWN,LOWER\_UP\> mtu 1500 qdisc fq\_codel state UP mode DEFAULT group default qlen 1000**
+**link/ether 80:fa:5b:6a:36:d2 brd ff:ff:ff:ff:ff:ff**
+
+**3: wlp0s20f3: \<BROADCAST,MULTICAST,UP,LOWER\_UP\> mtu 1500 qdisc noqueue state UP mode DORMANT group default qlen 1000**
+**link/ether 18:56:80:54:60:2d brd ff:ff:ff:ff:ff:ff**
+
+**
+
+Notice the indication of DOWN in the above output for the eth0 interface. This result means that Layer 1 isn't coming up. We might try troubleshooting by checking the cabling or the remote end of the connection (e.g., the switch) for problems.
+
+Before you start checking cables, though, it's a good idea to make sure that the interface isn't just disabled. Running a command to bring the interface up can rule this problem out:
+
+****\#*** *ip link* *set* *eth0 up**
+
+**
+
+The output of **ip link show** can be difficult to parse at a quick glance. Luckily, the **-br** switch prints this output in a much more readable table format:
+
+****\#*** *ip -br link show**
+
+**lo UNKNOWN 00:00:00:00:00:00 \<LOOPBACK,UP,LOWER\_UP\>**
+
+**eth0 UP 52:54:00:82:d6:6e \<BROADCAST,MULTICAST,UP,LOWER\_UP\>**
+
+It looks like **ip link set eth0 up** did the trick, and eth0 is back in business.
+
+These commands are great for troubleshooting obvious physical issues, but what about more insidious issues? Interfaces can negotiate at the incorrect speed, or collisions and physical layer problems can cause packet loss or corruption that results in costly retransmissions.
+
+How do we start troubleshooting those issues?
```

Comment:
Should this be a heading? i.e., ## How do we start troubleshooting those issues?

Reply:
changed